Assets/MRTK/Core/Utilities/Editor/Setup/MixedRealityToolkitFiles.cs

```diff
@@ -371,15 +410,21 @@ public static string GetGeneratedFolder
             }
         }
 
-        private static async Task SearchForFoldersAsync(string rootPath)
+        private static async Task SearchForFoldersAsync(List<string> rootFolders)
         {
             if (searchForFoldersToken != null)
             {
                 searchForFoldersToken.Cancel();
             }
 
             searchForFoldersToken = new CancellationTokenSource();
-            await Task.Run(() => SearchForFolders(rootPath, searchForFoldersToken.Token), searchForFoldersToken.Token);
+            await Task.Run(() =>
+                {
+                    for (int i = 0; i < rootFolders.Count; i++)
+                    {
+                        SearchForFolders(rootFolders[i], searchForFoldersToken.Token);
```

Comment:
Should we break if we ended up finding it (i.e. rather than keep searching)

Reply:
I'll take a look

Reply:
None of the methods in the call stack return or early exit as currently written. I will look at adding this.

Fortunately, the ones needing modification are all private. There will be no breaking changes involved.

Reply:
Oof yeah, none of the called methods provide any info on whether they were successful or not. Might be able to pipe a bool all the way back up, but I think some parts of this class need to be rewritten for robustness / performance anyway. Probably best to do it then.

Reply:
The scope of the required changes feel outside of that of this PR. There are some structural changes in MixedRealityToolkitFiles.cs that need investigation.